packages/console/src/Console.tsx

```diff
@@ -331,21 +331,34 @@ export class Console extends PureComponent<ConsoleProps, ConsoleState> {
           changes: VariableChanges;
         }
       | undefined,
-    historyItemParam: ConsoleHistoryActionItem,
+    historyItem: ConsoleHistoryActionItem,
     workspaceItemPromise: Promise<CommandHistoryStorageItem>
   ): void {
-    const historyItem = historyItemParam;
-    historyItem.wrappedResult = undefined;
-    historyItem.cancelResult = undefined;
+    const newHistoryItem = {
+      ...historyItem,
+      wrappedResult: undefined,
+      cancelResult: undefined,
+      result: result ?? historyItem.result,
+    };
+
+    this.setState(({ consoleHistory }) => {
+      const itemIndex = consoleHistory.lastIndexOf(historyItem);
+      if (itemIndex < 0) {
+        log.error(`historyItem not found in consoleHistory`);
+        return { consoleHistory };
```

Comment:
Do we want a re-render in this case? Might just return a `null` if we don't.

```suggestion
        return null;
```

Reply:
Sweet, I was trying to return `undefined` before and getting a TypeScript error - should've just returned `null` instead, I'm a bit of a moron. Good suggestion.